fix: [sc-54864] Adds safety check to provide near term fix to save query

eric-briscoe opened the pull request:
Adds safety check to provide near term fix to save query
<!---
Please write the PR title following the conventions at https://www.conventionalcommits.org/en/v1.0.0/
Example:
fix(dashboard): load charts correctly
-->

### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
There is an error when casting the columns array to String() for saving queries where the objects in the array are missing the toString method.   This is a near term rapid patch to fix workflow in production which will have a follow up to identify root cause.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
<!--- Skip this if not applicable -->

### TESTING INSTRUCTIONS
<!--- Required! What steps can be taken to manually verify the changes? -->
Repro steps:
1, go to sql lab and run query (click "RUN")
2, save query using save split button
3, observe confirmation msg
4, go to saved query list and look for save query

Expect:
user can save query using save split button

Actual:
user saw query can not be saved msg and the query did not show in save query list
### ADDITIONAL INFORMATION
<!--- Check any relevant boxes with "x" -->
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

eric-briscoe committed:
[sc-54864] Adds safety check to provide near term fix for Queries not saving

There is an error when casting the columns array to String() for saving queries where the objects in the array are missing the toString method.   This is a near term rapid patch to fix workflow in production which will have a follow up to identify root cause.

eschutho commented on superset-frontend/packages/superset-ui-core/src/connection/callApi/callApi.ts:
```diff
@@ -146,10 +146,23 @@ export default async function callApi({
+            } catch (e) {
+              // eslint-disable-next-line no-console
+              console.error(
+                `Unable to convert attribut '${key}' to a String(). ${key} is not part of the formData payload for call to ${url}`,
```
```suggestion
                `Unable to convert attribute '${key}' to a String(). ${key} is not part of the formData payload for call to ${url}`,
```

eric-briscoe committed:
fix typo

Co-authored-by: Elizabeth Thompson <eschutho@gmail.com>

eschutho reviewed (approved):
Looks good, with one small console typo.

eric-briscoe committed:
Adjusted chekc to be explicit for undefined instead of truthy

This fixes issue caught by unit test where the if statement was using a truthy check where it should be explicitly checking for value of undefined

eric-briscoe committed:
Adds new unit test to get 100% coverage for callApi

jinghua-qa commented:
/testenv up
